functions/assert_metadata_os.pp

```diff
@@ -0,0 +1,57 @@
+# Fails a compile if the client system is not compatible with the module's
+# metadata.json
+#
+# @param module_name
+#   The name of the module that should be checked
+#
+# @param match
+#   The level at which to match the release number
+#
+#   * none  -> No match on release (default)
+#   * full  -> Full release must match
+#   * major -> Only the major release must match
+#
+function simplib::assert_metadata_os (
```

Comment:
I think it should be `assert_os_support()`, because it is an assert function in the classic sense; it is also generic.

My biggest thing is this: the function name shouldn't know about the implementation, nor should the classes using this function.

Just imagine it like this: you add this function to a hundred modules, then sometime down the line you improve it so it can use some new feature to determine OS compatibility. You could keep using the same function name, but then doesn't make sense and creates cognitive dissonance in users who are trying to figure out why the function is failing their modules. You could also wrap the old function in a new one, but then you have all these old modules using the old function, not gaining the benefits of the new one.  Or, god forbid, you could go change the name in all those hundred plus odd modules-- sound familiar? On the other hand, you could start right now by using a generic name, then not have to worry about any of this later on.